planner/core/testdata/integration_serial_suite_out.json

```diff
@@ -1500,6 +1500,159 @@
       }
     ]
   },
+  {
+    "Name": "TestMppJoinDecimal",
+    "Cases": [
+      {
+        "SQL": "desc format = 'brief' select t1.c1, t1.c2, t2.c1, t2.c2, t2.c3 from t t1 join t t2 on t1.c1 + 1 = t2.c2 - 10 and t1.c1 * 3 = t2.c3 / 2",
+        "Plan": [
+          "Projection 12500.00 root  test.t.c1, test.t.c2, test.t.c1, test.t.c2, test.t.c3",
+          "└─TableReader 12500.00 root  data:ExchangeSender",
+          "  └─ExchangeSender 12500.00 cop[tiflash]  ExchangeType: PassThrough",
+          "    └─HashJoin 12500.00 cop[tiflash]  inner join, equal:[eq(Column#21, Column#22) eq(Column#15, Column#16)]",
+          "      ├─ExchangeReceiver(Build) 10000.00 cop[tiflash]  ",
+          "      │ └─ExchangeSender 10000.00 cop[tiflash]  ExchangeType: HashPartition, Hash Cols: Column#21, Column#15",
+          "      │   └─Projection 10000.00 cop[tiflash]  test.t.c1, test.t.c2, Column#13, Column#15, cast(Column#13, decimal(34,8) BINARY)->Column#21",
+          "      │     └─Projection 10000.00 cop[tiflash]  test.t.c1, test.t.c2, mul(test.t.c1, 3)->Column#13, plus(test.t.c1, 1)->Column#15",
```

Comment:
It seems that these two adjacent projections can be merged into one.

Reply:
`eliminatePhysicalProjection` may help you.
Or leave a TODO comment in `convertDecimalKeyIfNeed` and optimize this in future.

Reply:
Yeah, but we have called this optimize. I will leave a TODO.

Reply:
But Actually, It's no (performance) difference with TiFlash